index.js

```diff
@@ -66,7 +66,7 @@ module.exports = function kindOf(val) {
 };
 
 function ctorName(val) {
-  return val.constructor ? val.constructor.name : null;
+  return val.constructor && typeof val.constructor === 'function' ? val.constructor.name : null;
```

Comment:
We can get rid of `val.constructor && ` since the second condition covers this.

Reply:
Looks good to me. 
Thanks.

Reply:
Looks good to me.
Thanks.